packages/zen-bull-vault/test/e2e/ShutdownEmergencyWithdrawTest.t.sol

```diff
@@ -0,0 +1,257 @@
+import { IZenEmergencyWithdraw } from "../../src/interface/IZenEmergencyWithdraw.sol";
+import { StrategyMath } from "squeeth-monorepo/strategy/base/StrategyMath.sol";
+
+contract ShutdownEmergencyWithdrawTest is Test {
+    using StrategyMath for uint256;
+
+    address payable public constant ZEN_BULL = 0xb46Fb07b0c80DBC3F97cae3BFe168AcaD46dF507;
+    address public constant WETH = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+    address public constant USDC = 0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48;
+    address payable public constant CRAB = 0x3B960E47784150F5a63777201ee2B15253D713e8;
+    address public constant OSQTH = 0xf1B99e3E573A1a9C5E6B2Ce818b617F0E664E86B;
+    address public constant ETH_USDC_POOL = 0x8ad599c3A0ff1De082011EFDDc58f1908eb6e6D8;
+    address payable public constant CONTROLLER = 0x64187ae08781B09368e6253F9E94951243A493D5;
+    address public constant ORACLE = 0x65D66c76447ccB45dAf1e8044e918fA786A483A1;
+    address public constant ZEN_BULL_EMERGENCY_WITHDRAW = 0x3DdC956B08c0A6dA2249f8c528fF0594F5AEa381;
+
+    // multisig owner
+    address public constant OWNER = 0xAfE66363c27EedB597a140c28B70b32F113fd5a8;
+
+    uint256 public constant INDEX_SCALE = 1e4;
+    uint32 public constant TWAP_PERIOD = 420 seconds;
+
+    uint256 deployerPk;
+    uint256 user3Pk;
+    address deployer;
+    address user1;
+    address user2;
+    address user3;
+
+    ShutdownEmergencyWithdraw internal shutdownEmergencyWithdraw;
+
+    function setUp() public virtual {
+        string memory FORK_URL = vm.envString("FORK_URL");
+        vm.createSelectFork(FORK_URL, 21095588);
+
+        deployerPk = 0xA11CD;
+        deployer = vm.addr(deployerPk);
+        user1 = 0x73738c989398EBAfdAfD097836Fd910BAc14CCDC;
+        user2 = 0x1ba20876565280C8274eEd551B575936097e414F;
+        user3Pk = 0xB12CD;
+        user3 = vm.addr(user3Pk);
+
+        vm.label(deployer, "Deployer");
+        vm.label(user1, "User1");
+        vm.label(user2, "User2");
+        _deployAndConfigure();
+    }
+
+    function testShutdownEmergencyWithdrawNotOwner() public {
+        vm.startPrank(user1);
+        vm.expectRevert("Ownable: caller is not the owner");
+        shutdownEmergencyWithdraw.shutdownEmergencyWithdraw();
+        vm.stopPrank();
+    }
+
+    function testShutdownEmergencyWithdrawFromDeployerNotOwner() public {
+        vm.startPrank(deployer);
+        vm.expectRevert("Ownable: caller is not the owner");
+        shutdownEmergencyWithdraw.shutdownEmergencyWithdraw();
+        vm.stopPrank();
+    }
+
+    function testShutdownEmergencyWithdraw() public {
+        // get initial crab balance in zen bull strategy
+        uint256 initialZenBullCrabBalance = ZenBullStrategy(ZEN_BULL).getCrabBalance();
+
+        (uint256 ethInCrab, uint256 wPowerPerpInCrab) =
+            ZenBullStrategy(ZEN_BULL).getCrabVaultDetails();
+
+        uint256 ethShare = initialZenBullCrabBalance.wdiv(IERC20(CRAB).totalSupply());
```

Comment:
doesn't change anything - but this is more like crabShare

Reply:
changed to share